src/util/helpers.ts

```diff
@@ -20,15 +18,20 @@ export const andHandle = (error: Error) => {
 };
 
 /**
- * Merge destination and source recursively.
+ * Append entries from defaults which are missing from inputs.
  */
-export const mergeObjects = (destination: any, source: any) => {
-  const handleArrays = (objValue: any, srcValue: any) => {
-    if (Array.isArray(objValue) && Array.isArray(srcValue)) {
-      return srcValue;
+export const mergeObjects = (defaults: any, input: any) => {
+  const merged: Record<string, any> = {...input};
+
+  for (const key in defaults) {
+    if (Array.isArray(defaults[key])) {
+      merged[key] = input[key] !== undefined ? input[key] : defaults[key];
     }
-    return;
-  };
 
-  return mergeWith(destination, source, handleArrays);
+    if (!(key in input)) {
+      merged[key] = defaults[key];
+    }
+  }
+
+  return merged;
```

Comment:
Why not just use `{...defaults, ...inputs}` now?

Reply:
it won't do the trick since null values won't be overridden

Reply:
and that behavior should be covered by tests too, the current implementation doesn't satisfy that either